statsmodels/genmod/families/family.py

```diff
@@ -72,7 +75,14 @@ def _getlink(self):
     link = property(_getlink, _setlink, doc="Link function for family")
 
     def __init__(self, link, variance):
-        self.link = link()
+        if inspect.isclass(link):
```

Comment:
I looked at the stackoverflow question, but I think the problem is only with old style classes which we don't use. But I don't have a python 2 open right now to check